alert-common/src/main/java/com/synopsys/integration/alert/common/provider/Provider.java

```diff
@@ -24,12 +24,13 @@
 
 import java.util.List;
 
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
 import com.synopsys.integration.alert.common.provider.lifecycle.ProviderTask;
 import com.synopsys.integration.alert.common.provider.notification.ProviderDistributionFilter;
 import com.synopsys.integration.alert.common.provider.notification.ProviderNotificationClassMap;
 import com.synopsys.integration.alert.common.workflow.processor.ProviderMessageContentCollector;
 
-public abstract class Provider {
+public abstract class Provider<T extends ProviderProperties> {
```

Comment:
I think we can get away without this. The implementing class should be able to specify the correct type.